source/service.go

```diff
@@ -295,12 +294,35 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 				log.Errorf("Pod %s not found for address %v", address.TargetRef.Name, address)
 				continue
 			}
+			for _, container := range pod.Spec.Containers {
```

Comment:
we would need to guard to enable if you need it. We can not create more than 50k records in a bigger cluster.